app/controllers/api/base_controller/parser.rb

```diff
@@ -299,7 +299,7 @@ def validate_post_api_action_as_subcollection(cname, mname, aname)
         return if cname == @req.collection
         return if collection_config.subcollection_denied?(@req.collection, cname)
 
-        aspec = collection_config.typed_subcollection_actions(@req.collection, cname)
+        aspec = collection_config.typed_subcollection_actions(@req.collection, cname, @req.s_id ? :subresource : :subcollection)
```

Comment:
Up to this point we haven't needed to have a subresource POST action, so this wasn't needed in the previous PR